extensions/camel/camel-aws-s3/deployment/src/main/java/io/quarkus/camel/component/aws/s3/deployment/CamelAwsS3Processor.java

```diff
@@ -0,0 +1,99 @@
+                "org.apache.http.pool.ConnPoolControl", "com.amazonaws.http.conn.Wrapped");
+    }
+
+    @BuildStep(applicationArchiveMarkers = { "com/amazonaws" })
+    void preocess(CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
+            BuildProducer<ReflectiveMethodBuildItem> reflectiveMethod,
+            BuildProducer<SubstrateResourceBuildItem> resource,
+            BuildProducer<SubstrateResourceBundleBuildItem> resourceBundle,
+            BuildProducer<ServiceProviderBuildItem> serviceProvider,
+            ApplicationArchivesBuildItem applicationArchivesBuildItem) {
+
+        IndexView view = combinedIndexBuildItem.getIndex();
+
+        resource.produce(new SubstrateResourceBuildItem("com/amazonaws/partitions/endpoints.json"));
+        for (String s : getImplementations(view, JsonDeserializer.class)) {
+            reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, s));
+        }
+        for (String s : getImplementations(view, JsonSerializer.class)) {
+            reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, s));
+        }
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, Partitions.class.getCanonicalName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, Partition.class.getCanonicalName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, Endpoint.class.getCanonicalName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, Region.class.getCanonicalName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, Service.class.getCanonicalName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, CredentialScope.class.getCanonicalName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, LogFactoryImpl.class.getCanonicalName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, Jdk14Logger.class.getCanonicalName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, AWSS3V4Signer.class.getCanonicalName()));
+        reflectiveClass
+                .produce(new ReflectiveClassBuildItem(true, false, "com.sun.org.apache.xerces.internal.parsers.SAXParser"));
+        reflectiveClass
+                .produce(new ReflectiveClassBuildItem(true, false, "org.apache.camel.converter.jaxp.XmlConverter"));
+        reflectiveClass
+                .produce(new ReflectiveClassBuildItem(true, false, "com.github.benmanes.caffeine.cache.SSLMS",
+                        "com.github.benmanes.caffeine.cache.SILMS",
+                        "com.github.benmanes.caffeine.cache.PSMS",
+                        "com.github.benmanes.caffeine.cache.PDMS",
+                        "com.github.benmanes.caffeine.cache.SILMS"));
```

Comment:
These ones should be added to the Caffeine extension if they are not already there (I know some are there). And you should depend on the Caffeine extension.

Reply:
Ok, I'll add the extension